src/BSeries.jl

```diff
@@ -1952,16 +1953,18 @@ function _is_energy_preserving_dense(trees, coefficients)
 
     # We remove repeated columns
     unique!(energy_preserving_basis)
+    # Set the numerical tolerance
+    tolerance = 1e-8
 
     # The components of `energy_preserving_basis` are the columns of the matrix `M`.
     M = reduce(hcat, energy_preserving_basis)
-    rank_M = rank(M)
+    rank_M = rank(M, atol = tolerance)
 
     # We also create an extended matrix for which we append the vector of
     # coefficients
     Mv = [M coefficients]
-    rank_Mv = rank(Mv)
-
+    rank_Mv = rank(Mv, atol = tolerance)
```

Comment:
Why do you choose an absolute tolerance here instead of the default relative tolerance?

Reply:
For some methods obtained from Butcher arrays with floating inputs some information may be lost so that  the coefficients may not be exactly equal to the coefficient of their `energy_preserving_pair`. Then, for these methods it is better to set a tolerance to test if they are `energy-preserving` or not.

Reply:
Yes, I totally agree with this. What I was asking is whether it would make more sense to use a relative tolerance by default instead of an absolute one.

Reply:
Oh, I see. I didn't think about it. 
I can implement the relative tolerance instead of the absolute one if you prefer.